spa/src/components/common/RedirectWithReload.js

```diff
@@ -0,0 +1,9 @@
+import { useHistory, withRouter } from 'react-router-dom';
+
+const RedirectWithReload = (props) => {
+  let history = useHistory();
+  history.push({ pathname: props.to, search: props.location.search });
```

Comment:
Shouldn't we use `replace` instead of `push`?
`replace(path, [state]) - (function) Replaces the current entry on the history stack` 
`push(path, [state]) - (function) Pushes a new entry onto the history stack`

```suggestion
  history.replace({ pathname: to, search: location.search });
```

Reply:
Addressed with 02f12cff0cdaa189fe4b9b93678a7b00068cf129